spec/lib/rmt/downloader_spec.rb

```diff
@@ -101,6 +105,99 @@
+          drpm_file_1: downloader.download('/repo1/foo.drpm', checksum_type, drpm_content1_checksum),
+          drpm_file_2: downloader.download('/repo2/foo.drpm', checksum_type, drpm_content2_checksum) }
+      end
+
+      before do
+        stub_request(:get, 'http://example.com/repo1/foo.rpm')
+          .with(headers: headers)
+          .to_return(status: 200, body: rpm_content1, headers: {})
+        stub_request(:get, 'http://example.com/repo2/foo.rpm')
+          .with(headers: headers)
+          .to_return(status: 200, body: rpm_content2, headers: {})
+        stub_request(:get, 'http://example.com/repo1/foo.drpm')
+          .with(headers: headers)
+          .to_return(status: 200, body: drpm_content1, headers: {})
+        stub_request(:get, 'http://example.com/repo2/foo.drpm')
+          .with(headers: headers)
+          .to_return(status: 200, body: drpm_content2, headers: {})
+      end
+
+      it('creates rpm file from repo1') { expect(File.read(filenames[:rpm_file_1])).to eq(rpm_content1) }
+      it('creates rpm file from repo2') { expect(File.read(filenames[:rpm_file_2])).to eq(rpm_content2) }
+      it('does not duplicate rpm files') { expect(File.read(filenames[:rpm_file_1])).not_to eq(File.read(filenames[:rpm_file_2])) }
+      it('creates drpm file from repo1') { expect(File.read(filenames[:drpm_file_1])).to eq(drpm_content1) }
+      it('creates drpm file from repo2') { expect(File.read(filenames[:drpm_file_2])).to eq(drpm_content2) }
+      it('does not duplicate drpm files') { expect(File.read(filenames[:drpm_file_1])).not_to eq(File.read(filenames[:drpm_file_2])) }
+    end
+
+    context 'duplicated files' do
+      let(:checksum_type) { 'SHA256' }
+      let(:rpm_content) { 'foo' }
+      let(:rpm_content_checksum) { '2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae' }
+      let(:drpm_content) { 'dfoo' }
+      let(:drpm_content_checksum) { 'cebb0b202c71540e3300b551ec49b0e962c4931eb36ad25e4f1e277b9e767692' }
+      let(:filenames) do
+        { rpm_file_1: downloader.download('/repo1/foo.rpm', checksum_type, rpm_content_checksum),
+          rpm_file_2: downloader.download('/repo2/foo.rpm', checksum_type, rpm_content_checksum),
+          drpm_file_1: downloader.download('/repo1/foo.drpm', checksum_type, drpm_content_checksum),
+          drpm_file_2: downloader.download('/repo2/foo.drpm', checksum_type, drpm_content_checksum) }
+      end
+
+      before do
+        stub_request(:get, 'http://example.com/repo1/foo.rpm')
+          .with(headers: headers)
+          .to_return(status: 200, body: rpm_content, headers: {})
+        stub_request(:get, 'http://example.com/repo1/foo.drpm')
+          .with(headers: headers)
+          .to_return(status: 200, body: drpm_content, headers: {})
+      end
+
+      context 'handles duplicated files by copy' do
+        before do
+          deduplication_method(:copy)
```

Comment:
I would maybe add `.to have_been_made.once` to make sure it makes only one HTTP request, but that's not too critical.

Reply:
This was not worthwhile in the end, wontfix.